Runtime/Shaders/MapsBaker.cs

```diff
@@ -0,0 +1,230 @@
+
+        static Dictionary<string, ComputeShaderHelper> loadedShaders = new();
+
+        private const string SHADER_CONST_METALLIC = "Metallic";
+        private const string SHADER_CONST_SMOOTHNESS = "Smoothness";
+        private const string SHADER_CONST_RGB = "RGB";
+
+        private const string SHADER_FILE = "meshsync_channel_mapping";
+        private const string SHADER_NAME_SMOOTHNESS_INTO_ALPHA = "smoothness_into_alpha";
+        private const string SHADER_NAME_HDRP_MASK = "hdrp_mask";
+
+        
+        private static ComputeShaderHelper LoadShader(string name) {
+            string file = SHADER_FILE;
+
+            if (!loadedShaders.TryGetValue(file, out var shaderHelper)) {
+#if UNITY_EDITOR
+                var shaderFiles = AssetDatabase.FindAssets(file);
+                if (shaderFiles.Length > 0) {
+                    var shaderFile = shaderFiles[0];
+                    var shader = AssetDatabase.LoadAssetAtPath<ComputeShader>(AssetDatabase.GUIDToAssetPath(shaderFile));
+
+                    shaderHelper = new ComputeShaderHelper(shader, name);
+
+                    loadedShaders[name] = shaderHelper;
+                }
+#else
+// TODO: Load shader at runtime here
+#endif
+            }
+
+            return shaderHelper;
+        }
+
+        static MaterialPropertyData? FindMaterialProperty(List<MaterialPropertyData> materialProperties, string name) {
+            for (int i = 0; i < materialProperties.Count; i++) {
+                if (materialProperties[i].name == name) {
+                    return materialProperties[i];
+                }
+            }
+
+            return null;
+        }
+
+        private static bool FindTexture(string name,
+            List<TextureHolder> textureHolders,
+            List<MaterialPropertyData> materialProperties,
+            string fallbackPropertyName,
+            float fallbackPropertyValue,
+            out Texture2D texture) {
+            var prop = FindMaterialProperty(materialProperties, name);
+
+            if (prop.HasValue) {
+                MaterialPropertyData.TextureRecord rec = prop.Value.textureValue;
+                texture = BaseMeshSync.FindTexture(rec.id, textureHolders);
+
+                if (texture != null) {
+                    return true;
+                }
+            }
+
+            // If there is no such texture, create one with the given fallback value:
+
+            float value = fallbackPropertyValue;
+
+            for (int i = 0; i < materialProperties.Count; i++) {
+                if (materialProperties[i].name == fallbackPropertyName) {
+                    value = materialProperties[i].floatValue;
+                    break;
+                }
+            }
+
+            const int dim = 8;
+
+            var pixels = Enumerable.Repeat(new Color(value, value, value, value), dim * dim).ToArray();
+
+            texture = new Texture2D(dim, dim, UnityEngine.TextureFormat.RFloat, false, true);
```

Comment:
Need to ensure this is correctly disposed of

Reply:
This is a Texture2DDisposable now